mantidproject/mantid
# Move Run button to allow disabling

**Description of work.**
This PR moves the location of the Run button from the IndirectDataAnalysis Tab to be above the output section for each tab. This allows the run button to be easily disabled when a fit is being processed and then re-enabled afterwards.
This PR also disables the `Fit Single Spectrum` button in Indirect Data Analysis when a fit is running.

**To test:**
1. Go to `Interfaces`
2. Then `Indirect Data Analysis`
3. Make sure the `Run` button is above the output section in `Data Analysis`.
4. Enter the data below for the corresponding tabs
5. Click `Run`
6. Make sure the `Run` and `Fit Single Spectrum` buttons are disabled, and then re-enabled once the fit is complete

### **Test Data**
**Elwin Tab**
Input file: [irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2361048/irs26176_graphite002_red.zip)

**MSDFit Tab**
Input file: [osi92762_graphite002_elwin_eq.zip](https://github.com/mantidproject/mantid/files/2361050/osi92762_graphite002_elwin_eq.zip)
Fit Type: `Guassian`

**Iqt Tab**
Input files: [irs26173_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2361057/irs26173_graphite002_red.zip)
[irs26173_graphite002_res.zip](https://github.com/mantidproject/mantid/files/2361058/irs26173_graphite002_res.zip)


**IqtFit Tab**
Input file: [irs26176_graphite002_iqt.zip](https://github.com/mantidproject/mantid/files/2361062/irs26176_graphite002_iqt.zip)
Exponential: `2`
Note: drag the blue line on the graph to the left a bit

**ConvFit Tab**
Input files: [irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2361070/irs26176_graphite002_red.zip)
[irs26173_graphite002_res.zip](https://github.com/mantidproject/mantid/files/2361071/irs26173_graphite002_res.zip)
Fit Type: `One Lorenztian`

**F(Q) Tab**
Input file: [IN16B_125878_QLd_Result.zip](https://github.com/mantidproject/mantid/files/2361073/IN16B_125878_QLd_Result.zip)
Fit Type: `Chudley Elliott`

Fixes #22844 and #23223

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
